tests/python/test_copy_from.py

```diff
@@ -23,3 +23,32 @@ def test_scalar():
     assert y[0] == 1
     assert y[1] == 0
     assert y[2] == 3
+
+
+@test_utils.test()
+def test_struct():
+    @ti.dataclass
+    class C:
+        i: int
+        f: float
+
+    n = 16
+
+    x = C.field(shape=n)
+    y = C.field(shape=n)
```

Comment:
nit, add a test for 2d fields? since that was failing with a different error message

Reply:
That might not be necessary because they have the same root cause :-)